### ISSUE TITLE
TB Make - fos.cpp:21:10: fatal error - AWS Vanilla EC2 - ubuntu-bionic-18.04-amd64-server-20211027

### ISSUE THREAD

**Issue by idiom-bytes** [TrueBlocks/trueblocks-core#1876]
Hi there, I deployed a vanilla EC2 Ubuntu instance, and followed all the default initializing steps to bootstrap TB.

I understand that it's clearly throwing an exception while trying to include a filesystem, through libs/utillib/sfos.cpp, however this should likely be self-contained, and independent of anything that is local to my machine. Everything else (go, python, jq) seems to be working without a hitch.

Am still digging through things but am not sure why it's throwing this exception.

> [  8%] Building CXX object libs/utillib/CMakeFiles/util.dir/options_base_usage.cpp.o
> [  8%] Building CXX object libs/utillib/CMakeFiles/util.dir/performance.cpp.o
> [  8%] Building CXX object libs/utillib/CMakeFiles/util.dir/rpcresult.cpp.o
> [  9%] Building CXX object libs/utillib/CMakeFiles/util.dir/runtimeclass.cpp.o
> [  9%] Building CXX object libs/utillib/CMakeFiles/util.dir/sfarchive.cpp.o
> [  9%] Building CXX object libs/utillib/CMakeFiles/util.dir/sfos.cpp.o
> /home/ubuntu/trueblocks-core/src/libs/utillib/sfos.cpp:21:10: fatal error: filesystem: No such file or directory
>  #include <filesystem>
>           ^~~~~~~~~~~~
> compilation terminated.
> libs/utillib/CMakeFiles/util.dir/build.make:854: recipe for target 'libs/utillib/CMakeFiles/util.dir/sfos.cpp.o' failed
> make[2]: *** [libs/utillib/CMakeFiles/util.dir/sfos.cpp.o] Error 1
> CMakeFiles/Makefile2:411: recipe for target 'libs/utillib/CMakeFiles/util.dir/all' failed
> make[1]: *** [libs/utillib/CMakeFiles/util.dir/all] Error 2
> Makefile:129: recipe for target 'all' failed
> make: *** [all] Error 2

Here is my sample steps to bootstrap.

```
# inside remote ec2 instance
sudo apt update

# essential dependencies, doing a bunch of apt updates just to make sure that any dependencies / packages get updated after core installs
sudo apt install build-essential git cmake
sudo apt update
sudo apt install python3.8
alias python=python3
sudo apt install python-dev
sudo apt update
sudo apt install zlib1g-dev libncurses5-dev libgdbm-dev libnss3-dev libssl-dev libreadline-dev libffi-dev wget
sudo apt update
sudo apt install libcurl3-dev clang-format jq
sudo apt update

# from local - copy golang to remote
scp -i remote.pem go1.17.2.linux-amd64.tar.gz ubuntu@ec2-myipaddress:/home/ubuntu/

# back inside remote - remove & install go
rm -rf /usr/local/go
sudo tar -C /usr/local -xzf go1.17.2.linux-amd64.tar.gz

# configure bashrc, add path, source bashrc & verify
vim ~/.bashrc
export PATH=$PATH:/usr/local/go/bin
source ~/.bashrc
go version

# install trueblocks
git clone -b develop https://github.com/trueblocks/trueblocks-core
cd trueblocks-core
mkdir build && cd build
cmake ../src
make <<< THROWS EXCEPTION
```

Am wondering if anyone has any insights as to why `make` is throwing this exception.

**Comment by tjayrush** [comment]
I've never seen this before. I wonder if it would be possible for me to get into this machine via ssh? Can you connect with me in the discord? https://discord.gg/t27vSWsF

**Comment by tjayrush** [comment]
Also, here's our versions for the build on Ubuntu:

For `make --version`:
```
GNU Make 4.2.1
Built for x86_64-pc-linux-gnu
Copyright (C) 1988-2016 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
```

and for `cmake --version`:
```
cmake version 3.16.3

CMake suite maintained and supported by Kitware (kitware.com/cmake).
```

I'm pretty sure the problem you're seeing is with `make` or `cmake` (as opposed to Python or something).

**Comment by idiom-bytes** [comment]
@tjayrush thanks for sharing the make & cmake version expected.

I did a double take on everything and it became clear to me that the VM did not have the right versions installed. Upon looking further, I noticed that the versions expected are for 20.04 LTS, and thefore, utilizing an ubuntu 18.04 distro was not going to work.

I have now bootstrapped a 20.04 LTS VM, and have gotten past the compilation stage that was throwing the error.

I have no reason to think that compiling will fail at this point.

Thank you for your help, I am now closing this issue.
